sdk/go/auto/cmd.go

```diff
@@ -112,6 +112,7 @@ func NewPulumiCommand(opts *PulumiCommandOptions) (PulumiCommand, error) {
 	}
 
 	cmd := exec.Command(command, "version")
+	cmd.Env = append(os.Environ(), "PULUMI_SKIP_UPDATE_CHECK=true")
```

Comment:
I don't think we need the base environment here, `version` isn't looking at anything but the update check var

Reply:
Yeah, wasn't sure if there's any other contextual magic happening in there, so went with the conservative behavioural change.